torch/_dynamo/eval_frame.py

```diff
@@ -671,6 +672,15 @@ def __bool__(self):
             "you can specify them separately instead."
         )
 
+    @property
+    def serializable_spec(self):
+        return {
+            "t_id": self.t_id,
```

Comment:
For this to be truly useful in the serialized format, you need to know not an opaque id() of the tensor in question, but rather, which positional argument the constraints is being applied to

Reply:
OK. It looks like there are some confusion about why the constraints need to be serializable and how the serialized constraints will be used.

 1. The reason to save the constraints in a serializable format is just to not break the graph module serialization. The serialized format of the constraint itself doesn't matter. It will not be preserved after save/load.
 2. The saved constraints are used in memory directly after export to convert constraints to runtime assertion which is a separate pass after export.